File: contest/bigFib.py
""" bigFib.py """

def calcFibOld(limit):
    a = 1
    b = 1
    finNum = 0
    while finNum < limit:
       newNum = a + b
       a = b
       b = newNum
       finNum = sum(map(int,str(newNum)))
    return finNum

def calcFib(limit):
    a = 1
    b = 1
    sum = 0
    #print 1,1,
    for i in range(limit-2):
        sum = a + b
        a = b
        b = sum
        
        #print sum,
        
    #print
    return(sum)

def main():
    inFile = open("bigFibIn.txt", "r")
    for line in inFile:
        limit = int(line)
        print calcFib(limit)
        
    inFile.close()
    
if __name__ == "__main__":
    main()
    #print calcFib(5)

File: contest/bigFib.cpp
#include <iostream>
#include <fstream>

using namespace std;

unsigned long calcFib(int limit){
    unsigned long a = 1;
    unsigned long b = 1;
    unsigned long sum = 0;
    for (int i = 0; i < limit -2; i++){
        sum = a + b;
        a = b;
        b = sum;
    } // end for
    return(sum);
} // end calcFib

int main(){
    ifstream inFile;
    inFile.open("bigFibIn.txt");
    int limit;
    
    while(!inFile.eof()){
        inFile >> limit;
        cout << calcFib(limit) << endl;
    } // end while

    return(0);
} // end main